lib/src/presentation/widget/movie_screen/movie_genres/genres_section.dart

```diff
@@ -39,38 +43,60 @@ class _GenresSectionState extends State<GenresSection> {
           ),
           SizedBox(
             height: UiConstants.genresSizedBoxListViewMovieGenres,
-            child: FutureBuilder<List<String>>(
-              future: genres,
+            child: StreamBuilder<GenresEvent>(
+              stream: widget.genresStream,
               builder: (
                 BuildContext context,
-                AsyncSnapshot<List<String>> snapshot,
+                AsyncSnapshot<GenresEvent> snapshot,
               ) {
-                if (snapshot.hasData) {
-                  return ListView.builder(
-                    shrinkWrap: true,
-                    scrollDirection: Axis.horizontal,
-                    itemCount: snapshot.data!.length,
-                    itemBuilder: (
-                      BuildContext context,
-                      int index,
-                    ) {
-                      return GenresText(
-                        text: snapshot.data![index],
-                      );
-                    },
-                  );
-                } else if (snapshot.hasError) {
-                  return Text(
-                    snapshot.error.toString(),
-                  );
-                } else {
-                  return const Center(
-                    child: CircularProgressIndicator(),
-                  );
+                switch (snapshot.data?.state) {
+                  case BaseState.success:
+                    genres = GenresConverter().getGenresByIds(
+                      widget.genresIds,
+                      snapshot.data!.genreList,
+                    );
+                    return FutureBuilder<List<String>>(
+                      future: genres,
+                      builder: (
+                        BuildContext context,
+                        AsyncSnapshot<List<String>> snapshot2,
+                      ) {
+                        if (snapshot2.hasData) {
```

Comment:
Why this is call snapshot2, review the naming

Reply:
Deleted. As Mati said before the function getGenresById() is synchronous processing so no future constructor needed here anymore.